database/sql/V0029__create_blocks.sql

```diff
@@ -0,0 +1,5 @@
+CREATE TABLE blocks (
```

Comment:
I think this implementation has one disadvantage: it requires all future event_handlers for different events to be updated atomically.

Let me give an example: Imagine we will have another eventhandler that updates events from another contract. If this event handler would use the same blocks table, but the event handlers are not executed at the same time, the blocks table could be rewritten and accurate information for one eventhandler could be inaccurate for the other event handler. Hence, I think the better design is to not share the "blocks" table between event handlers. 

If we don't wanna share the blocks table per event handler, then I would go for the following way:

In every events table, we are currently storing (block_number, log_index). We would change the db designs to store (block_number, block_hash, log_index).

And then we implement last_block function like this:
```
pub async fn last_block(ex: &mut PgConnection) -> Result<i64, sqlx::Error> {
    const QUERY: &str = "\
            SELECT * from Values(
                (SELECT COALESCE(MAX(block_number), 0) as block_number, block_hash FROM trades ORDER BY block_number limit 1), \
                (SELECT COALESCE(MAX(block_number), 0), block_hash FROM settlements ORDER BY block_number limit 1), \
                (SELECT COALESCE(MAX(block_number), 0), block_hash FROM invalidations ORDER BY block_number limit 1), \
                (SELECT COALESCE(MAX(block_number), 0), , block_hash FROM presignature_events ORDER BY block_number limit 1)) order by block_number limit 1)
;";
    sqlx::query_scalar(QUERY).fetch_one(ex).await
pub async fn last_block(ex: &mut PgConnection) -> Result<(i64, BlockHash), sqlx::Error> {
    const QUERY: &str =
        "SELECT block_number, block_hash FROM blocks ORDER_BY block_number DESC LIMIT 1;";
    sqlx::query_as(QUERY).fetch_one(ex).await
}
```
Sorry, the query will not work, but hopefully one gets the idea